js/pro/wazirx.js

```diff
@@ -0,0 +1,607 @@
+            const market = this.market (symbol);
+            symbol = market['symbol'];
+        }
+        const url = this.urls['api']['ws'];
+        const messageHash = 'myTrades';
+        const message = {
+            'event': 'subscribe',
+            'streams': [ 'ownTrade' ],
+            'auth_key': token,
+        };
+        const request = this.deepExtend (message, params);
+        const trades = await this.watch (url, messageHash, request, messageHash);
+        if (this.newUpdates) {
+            limit = trades.getLimit (symbol, limit);
+        }
+        return this.filterBySymbolSinceLimit (trades, symbol, since, limit, true);
+    }
+
+    async watchOrderBook (symbol, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name wazirx#watchOrderBook
+         * @description watches information on open orders with bid (buy) and ask (sell) prices, volumes and other data
+         * @see https://docs.wazirx.com/#depth-stream
+         * @param {string} symbol unified symbol of the market to fetch the order book for
+         * @param {int|undefined} limit the maximum amount of order book entries to return
+         * @param {object} params extra parameters specific to the wazirx api endpoint
+         * @returns {object} A dictionary of [order book structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-book-structure} indexed by market symbols
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        symbol = market['symbol'];
+        const url = this.urls['api']['ws'];
+        const messageHash = 'orderbook:' + symbol;
+        const stream = market['id'] + '@depth';
+        const subscribe = {
+            'event': 'subscribe',
+            'streams': [ stream ],
+        }; // TODO check if replaces all streams
+        const request = this.deepExtend (subscribe, params);
+        const orderbook = await this.watch (url, messageHash, request, messageHash);
+        return orderbook.limit (limit);
+    }
+
+    handleDelta (bookside, delta) {
+        const bidAsk = this.parseBidAsk (delta, 0, 1);
+        bookside.storeArray (bidAsk);
+    }
+
+    handleDeltas (bookside, deltas) {
+        for (let i = 0; i < deltas.length; i++) {
+            this.handleDelta (bookside, deltas[i]);
+        }
+    }
+
+    handleOrderBook (client, message) {
+        //
+        //     {
+        //         "data": {
+        //             "E": 1659475095000,
+        //             "a": [
+        //                 ["23051.0", "1.30141"],
+        //             ],
+        //             "b": [
+        //                 ["22910.0", "1.30944"],
+        //             ],
+        //             "s": "btcusdt"
+        //         },
+        //         "stream": "btcusdt@depth"
+        //     }
+        //
+        const data = this.safeValue (message, 'data', {});
+        const timestamp = this.safeNumber (data, 'E');
```

Comment:
Let's not use `safeNumber` here, timestamps are not supposed to be ` floats`. We should always use `safeInteger` to parse ts